gradle/java/modules.gradle

```
@@ -0,0 +1,251 @@
+import java.util.jar.JarFile
+import java.util.stream.Collectors
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Configure other things required for the java module system layer.
+
+allprojects {
+  plugins.withType(JavaPlugin) {
+    // TODO, NOCOMMIT: Experiment on the morfologik subproject only, for now.
+    if (project.path == ":lucene:analysis:morfologik") {
```

```suggestion
    if (project.path in [":lucene:analysis:morfologik", ":lucene:demo"]) {
```
For experiments or tests, it could be useful since lucene/demo depends on lucene/expressions that does not has module-info.java for now (so `shouldBeOnModulePath()` returns false).

```
> Task :lucene:demo:showCompilationPaths
/mnt/hdd/repo/lucene/lucene/analysis/common/build/classes/java/main  <module-path>
/mnt/hdd/repo/lucene/lucene/core/build/classes/java/main  <module-path>
/mnt/hdd/repo/lucene/lucene/expressions/build/classes/java/main
/mnt/hdd/repo/lucene/lucene/facet/build/classes/java/main  <module-path>
/mnt/hdd/repo/lucene/lucene/queries/build/classes/java/main  <module-path>
/mnt/hdd/repo/lucene/lucene/queryparser/build/classes/java/main  <module-path>
/mnt/hdd/repo/lucene/lucene/sandbox/build/classes/java/main  <module-path>

```

---

Then the compilation failed... not sure this is the expected result; please ignore the suggestion if it does not make sense.
```
lucene $ ./gradlew -p lucene/demo/ compileJava -Duser.language=en

> Task :errorProneSkipped
WARNING: errorprone disabled (skipped on non-nightly runs)

> Task :lucene:demo:compileJava FAILED
/mnt/hdd/repo/lucene/lucene/demo/src/java-module/module-info.java:29: error: module not found: org.apache.lucene.expressions
  requires org.apache.lucene.expressions;
                            ^
1 error

FAILURE: Build failed with an exception.
```